compiler/rustc_resolve/src/lib.rs

```diff
@@ -1900,10 +1904,9 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     }
                 }
             }
-            let old_used = import.used.get();
-            let new_used = Some(used);
-            if new_used > old_used {
-                import.used.set(new_used);
+            let old_used = self.import_used_map.get(&import).copied();
+            if old_used < Some(used) {
+                self.import_used_map.insert(import, used);
```

Comment:
Not sure if it is important, but this can be done in one map lookup, something like:
```
let old_used = self.import_use_map.entry(&import).or_insert(used);
if *old_used < used {
  *old_used = used;
}
```